nUpdate.UpdateInstaller/WindowsEventLog.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Security;
+using System.Text;
+
+namespace nUpdate.UpdateInstaller
+{
+    public static class WindowsEventLog
+    {
+        private const string DefaultSource = "nUpdate";
+        private static string _Source;
+
+        /// <summary>
+        /// Gets or sets the source/caller.
+        /// </summary>
+        public static string Source
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_Source))
+                {
+                    _Source = DefaultSource;
+                    CreateEventSourceIfnotExsists();
+                }
+
+                return _Source;
+            }
+            set
+            {
+                _Source = value;
+                if (string.IsNullOrEmpty(_Source))
+                {
+                    _Source = DefaultSource;
+                }
+
+                CreateEventSourceIfnotExsists();
+            }
+        }
+        // https://stackoverflow.com/questions/25725151/write-to-windows-application-event-log
+        // Note: The actual limit is higher than this, but different Microsoft operating systems actually have different limits. So just use 30,000 to be safe.
+        private const int MaxEventLogEntryLength = 30000;
+
+
+        private static void CreateEventSourceIfnotExsists()

```

Comment:
There is a typo. Actually, I would prefer simply the name `CreateEventSource`.

Reply:
done